win32/tests/threadpool_win32_int/threadpool_win32_int.c

```diff
@@ -816,6 +821,20 @@ typedef struct CHAOS_TEST_DATA_TAG
 #define TIMER_PERIOD_MIN 50
 #define TIMER_PERIOD_MAX 400
 
+static void chaos_delay()
+{
+    // Earlier, the threadpool_open and threadpool_close would start and stop threadpool threads whereby there would be some limit on number of times work would be scheduled
+    // on Windows Threadpool using threadpool_schedule_work and threadpool_schedule_work_item because both functions would test for threadpool open condition and return if threadpool is not open.
+    // With threadpool_open and threadpool_close now only there for backward compatibility and doing nothing, the chaos test for optimized threadpool_schedule_work_item
+    // would create a very large number of work items compared to the previous version of threadpool_schedule_work. The tests would fail because expected_call_count would be very high than the
+    // executed_work_functions count because Windows Threadpool is unable to schedule that many work items. Adding Sleep(1) would significantly slow the tests. So adding following malloc and free
```

Comment:
>Windows Threadpool is unable to schedule that many work items [](http://example.com/codeflow?start=45&length=61)

This is interesting. Is there a specific limit here that we can reference?

Reply:
Removed this function for InterlockedHL Wait